js/client.js

```diff
@@ -1325,8 +1325,8 @@
 				if (this.curSideRoom === oldRoom) this.curSideRoom = room;
 				if (this.sideRoom === oldRoom) this.sideRoom = room;
 			}
-			if (type === BattleRoom) this.roomList.push(room);
-			if (type === ChatRoom || type === BattlesRoom) this.sideRoomList.push(room);
+			if (type === BattleRoom && !room.isSideRoom) this.roomList.push(room);
+			if (type === ChatRoom || type === BattlesRoom || (type === BattleRoom && room.isSideRoom)) this.sideRoomList.push(room);
```

Comment:
I've tried that, but it turns out that the Teambuilder, Ladder, Home and Rooms rooms are appended to the room lists separately, so they need to be excluded here or they'll get duplicate tabs.
I suppose there could be a "room is freely movable" flag for room objects, which includes only those room types that can be added to the room list in this way?